src/core/api/__tests__/public_api.test.ts

```diff
@@ -0,0 +1,561 @@
+
+        expect(player.setMaxBufferAhead(Infinity)).to.equal(undefined);
+        expect(player.getMaxBufferAhead()).to.equal(Infinity);
+      });
+    });
+
+    describe("getWantedBufferAhead/setWantedBufferAhead", () => {
+      it("should update the buffer goal through setWantedBufferAhead by default", () => {
+        expect(player.setWantedBufferAhead(50)).to.equal(undefined);
+        expect(player.getWantedBufferAhead()).to.equal(50);
+
+        expect(player.setWantedBufferAhead(Infinity)).to.equal(undefined);
+        expect(player.getWantedBufferAhead()).to.equal(Infinity);
+      });
+    });
+
+    describe("getAvailableAudioTracks", () => {
+      /* tslint:disable:max-line-length */
+      it("should return an empty array through getAvailableAudioTracks by default", () => {
+      /* tslint:enable:max-line-length */
+        expect(player.getAvailableAudioTracks()).to.eql([]);
+      });
+    });
+
+    describe("getAvailableTextTracks", () => {
+      it("should return an empty array through getAvailableTextTracks by default", () => {
+        expect(player.getAvailableTextTracks()).to.eql([]);
+      });
+    });
+
+    describe("getAvailableVideoTracks", () => {
+      /* tslint:disable:max-line-length */
+      it("should return an empty array through getAvailableVideoTracks by default", () => {
+      /* tslint:enable:max-line-length */
+        expect(player.getAvailableVideoTracks()).to.eql([]);
+      });
+    });
+
+    describe("getAudioTrack", () => {
+      it("should return undefined through getAudioTrack by default", () => {
+        expect(player.getAudioTrack()).to.equal(undefined);
+      });
+    });
+
+    describe("getTextTrack", () => {
+      it("should return undefined through getTextTrack by default", () => {
+        expect(player.getTextTrack()).to.equal(undefined);
+      });
+    });
+
+    describe("getVideoTrack", () => {
+      it("should return undefined through getVideoTrack by default", () => {
+        expect(player.getVideoTrack()).to.equal(undefined);
+      });
+    });
+
+    describe("setAudioTrack", () => {
+      it("should throw in setAudioTrack by default", () => {
+        expect(() => player.setAudioTrack("a")).to.throw();
+        expect(() => player.setAudioTrack("test")).to.throw();
+      });
+    });
+
+    describe("setTextTrack", () => {
+      it("should throw in setTextTrack by default", () => {
+        expect(() => player.setTextTrack("a")).to.throw();
+        expect(() => player.setTextTrack("test")).to.throw();
+      });
+    });
+
+    describe("setVideoTrack", () => {
+      it("should throw in setVideoTrack by default", () => {
+        expect(() => player.setVideoTrack("a")).to.throw();
+        expect(() => player.setVideoTrack("test")).to.throw();
+      });
+    });
+
+    describe("disableTextTrack", () => {
+      it("should disable text tracks in disableTextTrack by default", () => {
+        expect(player.disableTextTrack()).to.equal(undefined);
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Remove this use of the output from "disableTextTrack"; "disableTextTrack" doesn't return anything. ([typescript:S3699](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS3699))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=436&issues=AWgOFkKwUMkE2J58zqe2&open=AWgOFkKwUMkE2J58zqe2)

[ISSUE_KEY]: AWgOFkKwUMkE2J58zqe2